### PR TITLE
Issue/3434

### PR DESCRIPTION
**Pull request by ArqumFarooqui-NEC**
Included the section for `Special Metadata Types`

### CODE REVIEW

**Comment by fgalan** [comment]
Thank for your contribution!

I have had a look to the new section and I think it would be better to copy the same style than for the attributes built-in section. I'd suggest to copy-paste them and change only the differences (note that even in some casess the name of the built-in attribute matches the name of similar metadata, pe. dateCreated).

In addition, try to work on a cleaner way with regards the number of commits. Over-commiting in a PR is in general not a good idea, as it adds noise. This PR has now 15 commits and I wonder if it could be has been done in just one. Please, work on your working copy of the repository (not in github.com) and do all the changes derived of a review round all them in the same committ. Thanks!

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib [hunk 1]
```diff
@@ -354,6 +354,35 @@ The list of builtin attributes is as follows:
 Like regular attributes, they can be  used in `q` filters and in `orderBy`.
 However, they cannot be used in resource URLs.
 
+## Special Metadata Types
+
+Similar to attributes, each piece of metadata has a metadata name, a metadata type, a metadata value.
+The metadata field can also be used to explicitly include some special metadata e.g. dateCreated,
+dateModified and dateExpires i.e. for attribute creation date, last modification date and expiration date
+respectively (These are not included by default).
```
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib:362]:
```suggestion

Generally speaking, user-defined metadata types are informative; they are processed by the NGSIv2
server in an opaque way. Nonetheless, the types described below are used to convey a special
meaning:
```

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib [hunk 2]
```diff
@@ -354,6 +354,35 @@ The list of builtin attributes is as follows:
...
+```
+
+Datetimes are composed of date, time and timezone designator, in one of the following patterns:
+<date>
+<date>T<time>
+<date>T<time><timezone>
+
+Orion always provides datetime metadata using the format YYYY-MM-DDThh:mm:ss.ssZ. It uses UTC/Zulu timezone 
+(which is the best default option, as clients/receivers may be running in any timezone). 
+For more details about DateTime follow [DateTime section](https://fiware-orion.readthedocs.io/en/master/user/ngsiv2_implementation_notes/index.html)
```
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib:384]: By similarity with "Special Attribute Types" section, I'd suggest to delete this block of lines.

**Comment by fgalan** [comment]
> I have had a look to the new section and I think it would be better to copy the same style than for the attributes built-in section. I'd suggest to copy-paste them and change only the differences (note that even in some casess the name of the built-in attribute matches the name of similar metadata, pe. dateCreated).

I have provided more precise comments about this in the PR.

In addition, please add the following entry:

> Add "Special Metadata Types" section

just above "Changes since RC-2018.07:" line (for a future changelog, if needed).

Thanks!

**Comment by fgalan** [comment]
Something weird has happened with this PR... it shows no content in files changes...

I see a "force-pushed" was done. Be careful with that operation... in general it is not recomended. It can be dangerous.

**Comment by Chandradeep-NEC** [comment]
> Something weird has happened with this PR... it shows no content in files changes...
> 
> I see a "force-pushed" was done. Be careful with that operation... in general it is not recomended. It can be dangerous.

Dear @fgalan 

We are sorry for such commits in the PR. I'm assuring you as same will not be happen in future.

Thanks!!

**Comment by fgalan** [comment]
> We are sorry for such commits in the PR. I'm assuring you as same will not be happen in future.

Don't worry. Something it happens ;)

Thus, what's next? You will retry the same modifications again? Maybe it's a good idea to do it in a fresh new PR (in that case, please try to use only a small number of commits for the whole contribution)?

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib (hunk 1 above)
Comment (ArqumFarooqui-NEC) [reply doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib]: Fixed in 0a39e5b in PR #3654

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib (hunk 2 above)
Comment (ArqumFarooqui-NEC) [reply doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib]: Fixed in 0a39e5b in PR #3654

**Comment by ArqumFarooqui-NEC** [comment]
> In addition, please add the following entry:
> 
> > Add "Special Metadata Types" section
> 
> just above "Changes since RC-2018.07:" line (for a future changelog, if needed).
> 
Fixed in 0a39e5b in PR #3654 

Hi @fgalan sir, I have created a new PR #3654 , for issue #3434

**Comment by fgalan** [comment]
Ok, thanks. Thus, let's close this one and continue in PR #3654

**Pull request closed without merging**